### ISSUE TITLE
Confiurable Dynamic Y Axis Scale

### ISSUE THREAD

**Issue by pawanvjadhav** [highslide-software/highcharts.com#4248]
Hi,

We have bought license of Highstock.
We have observed following issue with scroll charts. We have communicated this issue with your support team over email. 

Scenario :
Draw a colum chart with scroll option true.
When the chart is scrolled horizontally the y axis scale changes dynamically. This gives the false implication of the height of bar  and y axis scale.

Sometimes this behavior(Dynamic Y Axis scale change) is useful if the values are spread across a large range of series values. This enables to see the scale depending upon the range available. Hence it should be configurable.

Example : http://jsfiddle.net/gq6v77hx/4/

Expected Result :
Dynamic scale change of y Axis on scrolling chart should be configurable.

We got reply from your support team of setting Min and Max.

But To set Min/ Max, we have to iterate through our data and find out Min and Max.
As per my understanding, when we enable Scroll, highchart re-calculate Min and Max from Cropped Dataset (visible). Our requirement is can you make this configurable?

In some case, change in bar height may mislead user while analysing chart, 
-Consider someone is looking at Sale of product by Year, as height of bar changes on scroll user has to keep eye on Y Axis scale. 

Is it possible for you to make this feature configurable?

**Comment by wergeld** [comment]
Why don't you let highcharts handle the max yAxis values in the load event like:
```
events: {
    load: function () {
        var points = []
        this.series.forEach(function (entry) {
            entry.data.forEach(function (theData) {
                points.push(theData.y);
            });
        });
        this.yAxis[0].update({
            max: Math.max.apply(Math, points)
        });
    }
}
```

See update jsFiddle: http://jsfiddle.net/wergeld/gq6v77hx/5/


This is a very basic example and you would need to make it more robust (like handling no series in chart).

**Comment by highslide-software** [comment]
@pawanvjadhav There's currently not an option for it, but there's actually a very simple hook you can use. Just add this line to make it work for all column charts and their derivatives:

```js
Highcharts.seriesTypes.column.prototype.getExtremesFromAll = true;
```
[View jsFiddle of the workaround](http://jsfiddle.net/highcharts/gq6v77hx/7/).

So the logic is already there, and we can simple expose this to an option. In the following commit, there's an option, `getExtremesFromAll`, added on the series level. It sits on the series because it is the series that reports its extremes back to the Y axis, and this option will make it report back not only the cropped data, but the whole range. [View jsFiddle of the implementation](http://jsfiddle.net/highcharts/gq6v77hx/8/).

**Comment by highslide-software** [comment]
Closed by f1b1bb1366064f0ddc9a58f7f1289baa1d50f94b.

**Comment by pawanvjadhav** [comment]
Thank You, I will try this.
